app/api/chemotion/reaction_api.rb

```diff
@@ -45,6 +45,121 @@ def rangebound(lower, upper)
     end
   end
 
+  def create_sub_sample(sample, current_user, reaction, fixed_label)
```

Comment:
Can you remove this code from the API and move this to a usecase object.
The api should be as lean as possible (the Grape Format gets even more unreadable with large chunks of code).

Reply:
@jhmegorei, to clarify, moving this to a usecase object means moving `update_materials_for_reaction` to `usecases/reactions/update_materials`?

```Ruby
module Usecases
  module Reactions
    class UpdateMaterials

    end
  end
end
```

Reply:
Hmm... I would recommend naming the class Usecases::Reactions::Update if you plan on naming the method `update_materials_for_reaction`, as otherwise the term `material` occurs both in the class name and the method.

While I would love to see the whole update process extracted into the usecase, it does not appear feasible at the moment, as there are more helpers involved (like ContainerHelper's `update_datamodel`), this is at least a step into the right direction.

Reply:
> While I would love to see the whole update process extracted into the usecase, it does not appear feasible at the moment, as there are more helpers involved (like ContainerHelper's `update_datamodel`), this is at least a step into the right direction.

@jhmegorei, agreed. I included the ContainerHelpers in the usecase for now (see latest commits).